crates/bevy_picking/src/mesh_picking/ray_cast/intersections.rs

```diff
@@ -94,17 +94,20 @@ pub fn ray_mesh_intersection<I: TryInto<usize> + Clone + Copy>(
 
             let triangle_index = Some(a);
             let tri_vertex_positions = &[
-                Vec3::from(positions[a]),
-                Vec3::from(positions[b]),
-                Vec3::from(positions[c]),
+                Vec3::from(*positions.get(a)?),
+                Vec3::from(*positions.get(b)?),
+                Vec3::from(*positions.get(c)?),
             ];
-            let tri_normals = vertex_normals.map(|normals| {
-                [
-                    Vec3::from(normals[a]),
-                    Vec3::from(normals[b]),
-                    Vec3::from(normals[c]),
-                ]
-            });
+            let tri_normals = vertex_normals
+                .map(|normals| -> Result<[Vec3; 3], ()> {
+                    Ok([
+                        Vec3::from(*normals.get(a).ok_or(())?),
+                        Vec3::from(*normals.get(b).ok_or(())?),
+                        Vec3::from(*normals.get(c).ok_or(())?),
+                    ])
+                })
+                .transpose()
+                .ok()?;
```

Comment:
This looks overcomplicated, I think this is equivalent, but more succinct:
```suggestion
            let tri_normals = vertex_normals.map(|normals| {
                Some([
                    Vec3::from(*normals.get(a)?),
                    Vec3::from(*normals.get(b)?),
                    Vec3::from(*normals.get(c)?),
                ])
            })?;
```

Reply:
That was my first instinct, but the options are swapped to how you'd expect them to behave.

I've changed it to a match which I think is more straightforward.

Reply:
> the options are swapped to how you'd expect them to behave

What do you mean? This should just be a simple `map` operation.

Reply:
Just to clarify, the return value of `vertex_normals.map()` will be `Option<Option<_>>`. Using `?` on the outer option will exit `ray_mesh_intersection` with `None` if `vertex_normals` is `None`, which is not the desired behaviour.